src/librustc_trans/mir/constant.rs

```diff
@@ -171,11 +128,11 @@ pub fn trans_static_initializer<'a, 'tcx>(
     let param_env = ty::ParamEnv::reveal_all();
     let static_ = cx.tcx.const_eval(param_env.and(cid))?;
```

Comment:
I'm thinking `const_eval` isn't the best, but it's probably an okay compromise.

Reply:
We could create a separate query for statics.